garden_ai/client.py

```diff
@@ -316,6 +315,12 @@ def build_container(self, pipeline: Pipeline) -> str:
         built_container_uuid = build_container(self.funcx_client, pipeline)
         return built_container_uuid
 
+    def register_pipeline(self, pipeline: Pipeline, container_uuid: str) -> str:
+        func_uuid = register_pipeline(self.funcx_client, pipeline, container_uuid)
+        pipeline.funcx_uuid = UUID(func_uuid)
```

Comment:
now that we can do it, shall we DOI it?

Reply:
```suggestion
        pipeline.funcx_uuid = UUID(func_uuid)
        self._mint_doi(pipeline)
```

Reply:
Oh yeah! Thanks for the reminder, this is definitely required behavior for pipeline registration.